src/lib.rs

```diff
@@ -115,6 +115,11 @@ pub trait OpenatDirExt {
     /// following the link.
     fn update_timestamps<P: openat::AsPath>(&self, path: P) -> io::Result<()>;
 
+    /// Update permissions for the given path (see `fchmodat(2)`).
+    ///
+    /// If the entry at `path` is a symlink, no action is performed.
```

Comment:
Hmm, shouldn't we just always follow symlinks? Since those perms are useless, following seems closer to what the client wants rather than a no-op. Or maybe better is to just make it a flag?

Reply:
The other helpers here are not following symlinks, that's why I initially went in that direction too.

I've scouted a bit of rpm-ostree and ostree logic, but most of the chmod-ing explicitly happens on non-symlinks. The only relevant hit I found is https://github.com/ostreedev/ostree/commit/08eaf668275b965d5a692ad48321a14f8c706360 where the consumer would like the `AT_SYMLINK_NOFOLLOW` behavior (but it can't).

Reply:
Ack OK. Maybe cleaner then to just leave it as is, and we can add a separate e.g. `set_mode_follow_symlink` down the line if needed.